modules/crypt.cpp

```diff
@@ -65,9 +199,17 @@ class CCryptMod : public CModule {
         AddCommand("ListKeys", static_cast<CModCommand::ModCmdFunc>(
                                    &CCryptMod::OnListKeysCommand),
                    "", "List all keys");
+        AddCommand("KeyX", static_cast<CModCommand::ModCmdFunc>(
+                                 &CCryptMod::OnKeyXCommand),
+                   "<Nick>", "Start a DH1080 key exchange with nick");
     }
 
-    ~CCryptMod() override {}
+    ~CCryptMod() override {
+        if (dh) {
+            DH_free(dh);
+            dh = NULL;
```

Comment:
Actually... I do not think you have to do anything in `CCryptMod`'s constructor. Instead, just use the following for assignment: `m_dh = std::unique_ptr<DH, void(*)(DH*)>(DH_new(), DH_free);`